### PR TITLE
Set draw state ibs

### PR DESCRIPTION
**Pull request by shanminchao**
Adding all set_draw_state IB packets as children of the set_draw_state packet:

image

### CODE REVIEW

File: dive_core/common/emulate_pm4.cpp [hunk 1]
```diff
@@ -860,7 +860,11 @@ bool EmulatePM4::QueueIB(uint64_t      ib_addr,
     // Grab current stack level info
     EmulateState::IbStack *cur_ib_level = emu_state->GetCurIb();
 
-    if (cur_ib_level->m_ib_queue_size >= EmulateState::kMaxPendingIbs)
+    // Queue up only non-0-sized IBs
```
Comment (wangra-google) [inline dive_core/common/emulate_pm4.cpp:863]: in what case we have 0 sized ibs?

File: dive_core/command_hierarchy.cpp [hunk 2]
```diff
@@ -755,6 +759,19 @@ bool CommandHierarchyCreator::OnIbStart(uint32_t                  submit_index,
         ib_string_stream << "DrawState IB"
                          << ", Address: 0x" << std::hex << ib_info.m_va_addr
                          << ", Size (DWORDS): " << std::dec << ib_info.m_size_in_dwords;
+
+        // Make all subsequent shared node parent be the relevant group in the set_draw_state packet
```
Comment (wangra-google) [inline dive_core/command_hierarchy.cpp:763]: sorry still a bit confused: what is shared node parent?

**Review by wangra-google — APPROVED** [review]

File: dive_core/command_hierarchy.cpp (hunk 2 above)
Comment (shanminchao) [reply dive_core/command_hierarchy.cpp:763]: A "shared node" is a node that can have multiple parents (i.e. shared among many different topologies). This is used exclusively for nodes that represent PM4 packets. PM4 packets can be children of IB-nodes, or children of other shared-nodes (e.g. children of INDIRECT_BUFFER packets), or children of draw call nodes, or even children of submit-nodes. So we needed to treat them a little differently.

File: dive_core/common/emulate_pm4.cpp (hunk 1 above)
Comment (wangra-google) [reply dive_core/common/emulate_pm4.cpp:863]: good to know, thanks

File: dive_core/command_hierarchy.cpp (hunk 2 above)
Comment (wangra-google) [reply dive_core/command_hierarchy.cpp:763]: aha, thanks!

**Pull request merged**